src/wp-includes/media.php

```diff
@@ -1058,13 +1058,19 @@ function wp_get_attachment_image( $attachment_id, $size = 'thumbnail', $icon = f
 		 * @param string $context The context. Default 'wp_get_attachment_image'.
 		 */
 		$context = apply_filters( 'wp_get_attachment_image_context', 'wp_get_attachment_image' );
+		$attr    = wp_parse_args( $attr, $default_attr );
```

Comment:
We were trying to avoid array_merges here. Why was this added again?

Reply:
@spacedmonkey This is something that was originally there (see [here](https://github.com/WordPress/wordpress-develop/pull/4495/files#diff-4d28b6652f0dab208c19aa45e1cf09f3c967060d8cc77502bfcd473a3216b612L1067)). We only moved the default loading logic above it to the newer logic bellow it.